chat-plugins/guesstheanime.js

```diff
@@ -0,0 +1,258 @@
+						}
+						this.answers = this.answers.filter((ans, i, arr) => {
+							return i === arr.map(toId).indexOf(toId(ans));
+						});
+
+						//anilist's api's kinda shit, so I'm gonna have to add these manually :/
+						//I might create commands for anime aliases sooner or later
+						if (this.answers.includes('Mahou Shoujo Madoka★Magica')) {
+							this.answers.unshift('Madoka Magica');
+						} else if (this.answers.includes('CODE GEASS: Hangyaku no Lelouch')) {
+							this.answers.unshift('Code Geass');
+						} else if (this.answers.includes('Higurashi no Naku Koro ni')) {
+							this.answers.unshift('Higurashi');
+						} else if (this.answers.includes('Gyakkyou Burai Kaiji: Ultimate Survivor')) {
+							this.answers.unshift('Gyakkyou Burai Kaiji');
+						} else if (this.answers.includes('The Melancholy of Haruhi Suzumiya')) {
+							this.answers.unshift('Haruhi Suzumiya');
+							this.answers.unshift('Haruhi');
+						} else if (this.answers.includes('Kino no Tabi: The Beautiful World')) {
+							this.answers.unshift('Kino no Tabi');
+						} else if (this.answers.includes('Assassination Classroom')) {
+							this.answers.unshift('AssClass');
+						} else if (this.answers.includes('Neon Genesis Evangelion')) {
+							this.answers.unshift('NGE');
+							this.answers.unshift('Evangelion');
+						}
+					})
+				);
+			})
+		);
+	}
+	update(self) {
+		let pointTotal = [];
+		this.points.forEach((points, user) => {
+			pointTotal.push('<i>' + user.name + ':</i> <span style = "color:green">' + points + '</span>');
+		});
+		let msg = '|html|<div class = "infobox"><center><b><i>Points:</i></b> ' + (pointTotal.length ? pointTotal.join(", ") : '<i>None</i>') + '<br>' + 'Character: <b>' + this.char + '</b></center>';
+		if (self) return self.sendReply(msg);
+		this.room.add(msg).update();
+	}
+	proceed() {
+		this.canGuess = false;
+		this.getAnime().then(() => {
+			this.canGuess = true;
+			this.guesses = new Map();
+			this.resetTimer();
+			this.update();
+		}).catch(err => {
+			console.log(err);
+			this.room.add('This game of Guess the Anime has been ended due to an error.').update();
+			this.end();
+		});
+	}
+	guess(user, guess, self) {
+		if (!guess || !guess.trim()) return self.sendReply("/guessanime [anime name] - Guesses the anime a character is from.");
+		if (!this.canGuess) return self.errorReply("This round is already over.");
+		if (this.guesses.get(user) >= 2) return self.sendReply("You've already guessed twice this round!");
+		if (guess.length > 60) return self.sendReply("Your answer cannot be more than 60 characters long.");
+		if (this.guessed.includes(toId(guess))) return self.sendReply("'" + guess + "' has already been guessed.");
+
+		if (this.answers.map(toId).includes(toId(guess))) {
+			this.room.add('|html|<b style = "color:' + Wisp.hashColor(user.userid) + ';">' + Tools.escapeHTML(user.name) + '</b> guessed <b>' + guess + '</b>, which was the correct answer!');
```

Comment:
unless the method's being called inside a callback function, <code>.update()</code> isn't necessary, but okie